skynet/modules/ttt/openai_api/app.py

```diff
@@ -23,16 +26,28 @@ def initialize():
 
     global proc
 
-    proc = subprocess.Popen(
-        f'{openai_api_server_path} \
-            --batch-size {llama_n_batch} \
-            --ctx-size {llama_n_ctx} \
-            --flash-attn \
-            --model {llama_path} \
-            --n-gpu-layers {llama_n_gpu_layers} \
-            --port {openai_api_server_port}'.split(),
-        shell=False,
-    )
+    if use_vllm:
+        openai_api_server_path = vllm_server_path
+        proc = subprocess.Popen(
+            f'python -m {openai_api_server_path} \
+                --model {llama_path} \
+                --gpu_memory_utilization 0.95 \
+                --max-model-len {llama_n_ctx} \
+                --port {openai_api_server_port}'.split(),
```

Comment:
Now that I see this, is it possible to limit the interface to localhost?

Reply:
Not sure I understand this correctly but we could remove the [openai_api_base_url](https://github.com/jitsi/skynet/blob/master/skynet/env.py#L40) from being exposed as an env variable

Reply:
So, when you run this, does it bind to 0.0.0.0:PORT or to 127.0.0.1:PORT? The former would expose the service to the internet, if there is no port filtering, the latter won't.



> we could remove the [openai_api_base_url](https://github.com/jitsi/skynet/blob/master/skynet/env.py#L40) from being exposed as an env variable

I'd keep it because it makes testing other models very easy if you run ollama locally for example and put the URL in there...

Reply:
In fact, perhaps we should think about that as the "development" option rather than running llama-server. That's a different discussion though.

Reply:
Awesome!